src/test/java/com/example/demo/DemoApplicationTests.java

```diff
@@ -1,13 +1,88 @@
 package com.example.demo;
 
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+import static org.assertj.core.api.AssertionsForClassTypes.assertThatExceptionOfType;
+
+
+import com.example.demo.dto.ProductRequestDto;
+import com.example.demo.dto.ProductResponseDto;
 import org.junit.jupiter.api.Test;
 import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.web.server.LocalServerPort;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.client.RestClient;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+public class DemoApplicationTests {
+
+  @LocalServerPort
+  private int port;
+
+  private RestClient client = RestClient.builder().build();
+
+  @Test
+  void 유효한_상품정보로_생성하면_201이_반환된다(){
+    ProductRequestDto dto = new ProductRequestDto("새상품", 1000, "http://image.com/1.jpg");
+    String url = "http://localhost:" + port + "/products";
+    ResponseEntity<ProductResponseDto> response = client.post()
+        .uri(url)
+        .body(dto)
+        .retrieve()
+        .toEntity(ProductResponseDto.class);
+
+    assertThat(response.getStatusCode()).isEqualTo(HttpStatus.CREATED);
+  }
+  @Test
+  void 전체_상품을_조회하면_200이_반환된다() {
+    String url = "http://localhost:" + port + "/products";
+
+    ResponseEntity<ProductResponseDto[]> response = client.get()
+                                                          .uri(url)
+                                                          .retrieve()
+                                                          .toEntity(ProductResponseDto[].class);
 
-@SpringBootTest
-class DemoApplicationTests {
+    assertThat(response.getStatusCode()).isEqualTo(HttpStatus.OK);
+    assertThat(response.getBody()).isNotNull();
+  }
 
   @Test
-  void contextLoads() {
+  void 존재하는_상품아이디로_조회하면_200과_상품정보가_반환된다(){
+    String url = "http://localhost:" + port + "/products/1";
+
+    ResponseEntity<ProductResponseDto> response = client.get()
+                                                        .uri(url)
+                                                        .retrieve()
+                                                        .toEntity(ProductResponseDto.class);
+
+    assertThat(response.getStatusCode()).isEqualTo(HttpStatus.OK);
+    assertThat(response.getBody()).isNotNull();
```

Comment:
```suggestion
    assertAll(
                () -> assertThat(response.getStatusCode()).isEqualTo(HttpStatus.OK),
                () -> assertThat(response.getBody()).isNotNull()
        );
```
이렇게도 묶을 수 있답니다😃 
assertAll()로 묶으면 어떤 이점이 있을지는 학습해보시죠..! 사용해보면서 불편한점도 같이 느끼면 베스트👍

Reply:
전자는 가독성이 뛰어나지만, 위쪽의 한 개가 먼저 실패를 한다면 두 번째 테스트는 실행되지 않는다는 점에서 단점이 존재하는 것 같습니다. 하지만 후자는 한 개가 실패하더라도 각각이 독립적으로 시행되기 때문에 두 번째 테스트는 실행된다는 점에서 큰 이점이 존재하는 것 같습니다!!